File: examples/chapter07/svdproperties.m
%%
% We verify some of the fundamental SVD properties using the built-in
% MATLAB command |svd|.
A = vander(1:5);
A = A(:,1:4)

%%
[U,S,V] = svd(A);
norm(U'*U - eye(5))

%%
norm(V'*V - eye(4))

%%
sigma = diag(S)

%%
[ norm(A) sigma(1) ]

%%
[ cond(A) sigma(1)/sigma(4) ]

File: julia/scripts/chapter07/svdproperties.jl

A = [i^j for i=1:5, j=0:3]

using LinearAlgebra
U,sigma,V = svd(A);

@show size(U),opnorm(U'*U - I);

@show size(V),opnorm(V'*V - I);

sigma

@show opnorm(A),sigma[1];

@show cond(A), sigma[1]/sigma[end];
